### ISSUE TITLE
Use epoll as default reactor on Linux

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5478]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5478|
|Type|enhancement|
|Created|2012-02-10 14:51:01Z|
|Branch|https://github.com/twisted/twisted/tree/epoll-default-5478|

epoll is most scalable event loop on Linux; it should be the default. 

This depends on [#4429](https://github.com/twisted/twisted/issues/4429) being fixed, sadly, otherwise we'll randomly break working applications that used twisted.internet.stdio.

<details><summary>Searchable metadata</summary>

```
trac-id__5478 5478
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Twisted_12_1 Twisted-12.1
branch__branches_epoll_default_5478 branches/epoll-default-5478
branch_author__itamar itamar
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1328885461000000 1328885461000000
changetime__1336435451000000 1336435451000000
version__None None
owner__itamar itamar
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @itamarst|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [34331]) Branching to 'epoll-default-5478'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst removed owner|
|-|-|

Ready for review. Note that this can only be merged after [#4429](https://github.com/twisted/twisted/issues/4429) is merged.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/epoll-default-5478

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

The choosing-reactor howto mentions epoll for linux 2.6 a couple of times, but it's supported on newer versions as well of course.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @itamarst|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set status to `closed`|
|-|-|

(In [34355]) Merge epoll-default-5478: epoll is now default reactor on Linux.

Author: itamar
Review: therve
Fixes: [#5478](https://github.com/twisted/twisted/issues/5478)

epoll is the default on linux; if not available, poll is used. Also, platforms where poll was default will now fall back to select.
